mapclientplugins/geometricfitstep/view/geometricfitwidget.py

```diff
@@ -453,207 +482,287 @@ def _displaySurfacesTranslucentClicked(self):
 
-    def _getConfig(self):
-        config = self._currentFitterStep
-        assert isinstance(config, FitterStepConfig)
-        return config
-
-    def _updateConfigWidgets(self):
-        """
-        Update config widgets to display settings for Fitter.
+    def _updateGroupSettingWidgets(self):
         """
-        config = self._getConfig()
-        self._ui.configProjectionCentreGroups_checkBox.setCheckState(QtCore.Qt.Checked if config.isProjectionCentreGroups() else QtCore.Qt.Unchecked)
-        self._updateDataProportion()
-        self._updateDataWeight()
-
-    def _configModelCoordinatesFieldChanged(self, index):
-        """
-        Callback for change in model coordinates field chooser widget.
+        Update and display group setting widgets for currentFitterStep
         """
-        field = self._ui.configModelCoordinates_fieldChooser.getField()
-        if field:
-            self._fitter.setModelCoordinatesField(field)
+        #isAlign = isinstance(self._currentFitterStep, FitterStepAlign)
+        isConfig = isinstance(self._currentFitterStep, FitterStepConfig)
+        isFit = isinstance(self._currentFitterStep, FitterStepFit)
+        if isConfig:
+            self._updateCentralProjection()
+            self._updateDataProportion()
+        elif isFit:
+            self._updateDataWeight()
+            self._updateStrainPenalty()
+            self._updateCurvaturePenalty()
+        self._ui.configCentralProjection_checkBox.setVisible(isConfig)
+        self._ui.configSetCentralProjection_checkBox.setVisible(isConfig)
+        self._ui.configDataProportion_checkBox.setVisible(isConfig)
+        self._ui.configDataProportion_lineEdit.setVisible(isConfig)
+        self._ui.configDataWeight_checkBox.setVisible(isFit)
+        self._ui.configDataWeight_lineEdit.setVisible(isFit)
+        self._ui.configStrainPenalty_checkBox.setVisible(isFit)
+        self._ui.configStrainPenalty_lineEdit.setVisible(isFit)
+        self._ui.configCurvaturePenalty_checkBox.setVisible(isFit)
+        self._ui.configCurvaturePenalty_lineEdit.setVisible(isFit)
 
-    def _configDataCoordinatesFieldChanged(self, index):
+    def _configSettingGroupChanged(self, index):
         """
-        Callback for change in data coordinates field chooser widget.
+        Callback for change in setting group field chooser widget.
         """
-        field = self._ui.configDataCoordinates_fieldChooser.getField()
-        if field:
-            self._fitter.setDataCoordinatesField(field)
+        self._updateGroupSettingWidgets()
 
-    def _configMarkerGroupChanged(self, index):
-        """
-        Callback for change in marker group field chooser widget.
-        """
-        group = self._ui.configMarkerGroup_fieldChooser.getField()
+    def _getGroup(self):
+        group = self._ui.configSettingGroup_fieldChooser.getField()
         if group:
-            self._fitter.setMarkerGroup(group)
+            group = group.getName()
+        return group
```

Comment:
This should be getGroupSettingsGroupName(self), and return groupName, not group (latter would be a Zinc FieldGroup object). Update where it's called and variable names in client code.